### PR TITLE
v0.1.0 RC

### PR DESCRIPTION
**Pull request by iky**

### CODE REVIEW

File: src/ddebounce.py [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+        @wrapt.decorator
+        def wrapper(wrapped, instance, args, kwargs):
+            key = format_key(*args, **kwargs)
+            if self.acquire(key):
+                return wrapped(*args, **kwargs)
+
+        return wrapper(wrapped)
+
+
+def debounce(lock, wrapped=None, key=None, repeat=False, callback=None):
```
Comment (timbu) [inline src/ddebounce.py]: just wondering if the `lock` argument here is the right name...  The value of this must be a redis client (or attrgetter for a redis client - is that correct?

File: src/ddebounce.py [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+
+    @wrapt.decorator
+    def wrapper(wrapped, instance, args, kwargs):
+        if instance and isinstance(lock, operator.attrgetter):
+            decorated = Lock(lock(instance)).debounce(
+                wrapped, key, repeat, callback)
+        else:
+            decorated = Lock(lock).debounce(
```
Comment (timbu) [inline src/ddebounce.py]: do we need a way to modify the `default_ttl` when instantiating the `Lock`? (I don't think we had this in the old version though btw 😄 )

**Comment by timbu** [comment]
very cool - 👍

File: src/ddebounce.py (hunk 1 above)
Comment (iky) [reply src/ddebounce.py]: :+1:  Good point, it's the redis client used then by lock instance

File: src/ddebounce.py (hunk 2 above)
Comment (iky) [reply src/ddebounce.py]: Yeah, nice enhancement suggestion .)

**Comment by timbu** [comment]
cool 👍

**Pull request merged**